accepted/2024/package-update-command.md

```diff
@@ -0,0 +1,266 @@
+# dotnet package update command
+
+- [Olia Gavrysh](https://github.com/OliaG/)
+- Start Date (2024-11-11)
+- [#13372](https://github.com/NuGet/Home/issues/13372)
+- [#4103](https://github.com/NuGet/Home/issues/4103)
+
+## Summary
+
+<!-- One-paragraph description of the proposal. -->
+The `dotnet package update` feature is designed to enhance the user experience by enabling .NET developers to address security vulnerabilities, deprecated packages, and outdated versions of their NuGet dependencies with a single command.
+This improvement will, in turn, bolster the overall security and reliability of .NET applications. The feature will build upon the existing NuGet Audit functionality.
+
+## Motivation
+
+<!-- Why are we doing this? What pain points does this solve? What is the expected outcome? -->
+Even though .NET developers now have the opportunity to learn about vulnerabilities and deprecated dependencies in their NuGet packages and addressing those issues, there are gaps in user experience, especially in the CLI workflows.
+
+### Existing capabilities
+
+Visual Studio
+
+- Has a means to show outdated packages
+- Has a means to show deprecate/vulnerable packages
+- Has a means to update all packages to latest
+- Has a means to update a single package to latest (which can be challenging)
+- Has a vulnerability filter, which in turn can allow you to update all vulnerable packages to latest
+dotnet
+
+CLI
+
+- Has a means to show outdated packages
+- Has a means to show deprecated/vulnerable packages
+- Has a means to update a single package
+- Does not have means to update multiple packages at once
+
+### Gaps in CLI experience
+
+- No means to update to all latest packages on the CLI
+- No means for update heuristics, such as, update my vulnerable packages
+
+This proposal is focused on bringing the CLI experience up to funcional parity with the experience in Visual Studio and make developers more productive by providing a single command to update all packages to the latest version and a way to update all vulnerable packages.
+
+## User experience
+
+### Updating all packages to the latest version
+
+When users want to ensure their NuGet dependencies are up to date, they can run a command in .NET CLI `dotnet package update` for a project or a solution.
+
+```CLI
+C:\> dotnet package update [<SOLUTION_PATH>|<PROJECT_PATH_>]
```

Comment:
It's helpful to also have a section that lists all the command arguments & options in one place, similar to what `dotnet package update -h` will show, because if both sln/proj and also package names are optional `dotnet package update [sln|proj] [package id]`, then it gets ambiguous when a single argument is provided. Does `dotnet package update Contoso.Utilities` mean update all package in a project with that name, or update that package to the latest version in whatever project/sln file happens to be in the current directory?

When a single argument is provided, we can do a Directory.Exists and File.Exists to guess, but I feel like that isn't great.  I think a `--project <path>` argument, which can accept a directory or sln file too, is my preferred design.